atlas_densities/densities/mtype_densities_from_map/create.py

```diff
@@ -73,62 +70,38 @@ def create_from_probability_map(
             - the sum of each row of the probability map is different from 1.0
             - the labels of the rows and columns of `probablity_map` are not all lowercase
             or contain some white spaces.
-            - the layer names appearing in the row labels of `probability_map` are not those
-            described in `metadata`
     """
     # pylint: disable=too-many-locals
-    region_ids = []
-    for region in metadata["regions"]:
-        region_ids.extend(
-            [
-                region_id
-                for region_id in region_map.find(
-                    region["query"], region["attribute"], with_descendants=True
-                )
-                if region_map.is_leaf_id(region_id)
-            ]
-        )
-    region_acronyms = [region_map.get(region_id, "acronym") for region_id in region_ids]
-
-    regions_acronyms_diff, molecular_types_diff = _check_probability_map_consistency(
-        probability_map, set(region_acronyms), set(molecular_type_densities.keys())
-    )
-    region_ids = [
-        region_id
-        for region_acronym, region_id in zip(region_acronyms, region_ids)
-        if region_acronym not in regions_acronyms_diff
-    ]
-    region_acronyms = [
-        region_acronym
-        for region_acronym in region_acronyms
-        if region_acronym not in regions_acronyms_diff
-    ]
+    region_info = region_map.as_dataframe().reset_index().set_index('acronym')
+    region_info = region_info.loc[probability_map.index.get_level_values('region')]
+    region_info = region_info.reset_index()[['region', 'id']]
```

Comment:
it is much shorter but the readability is extremely low in my opinion, so I would rather find some sort of middle ground

Reply:
It's pretty standard pandas usage no?

Reply:
One could also do this, I suppose:
```
region_info = (region_map
                          .as_dataframe()
                          .reset_index()
                          .set_index('acronym')
                          .loc[probability_map.index.get_level_values('region')]
                          .reset_index()[['region', 'id']]
)
```